src/django_components/component_media.py

```diff
@@ -730,96 +753,157 @@ def _resolve_component_relative_files(
-                    f"Interpreting template '{repr(filepath)}' of component '{module_name}'"
-                    " relatively to component file"
-                )
+    def resolve_relative_media_file(
+        filepath: Union[str, SafeData],
+        allow_glob: bool,
+    ) -> List[Union[str, SafeData]]:
+        resolved_filepaths, has_matched = resolve_media_file(
+            filepath,
+            allow_glob,
+            static_files_dir=matched_component_dir_abs,
+            media_root_dir=component_module_dir_path_abs,
+        )
 
-                return filepath_rel_to_comp_dir
+        # NOTE: It's important to use `repr`, so we don't trigger __str__ on SafeStrings
+        if has_matched:
+            logger.debug(
+                f"Interpreting file '{repr(filepath)}' of component '{module_name}'" " relatively to component file"
+            )
+        else:
+            logger.debug(
+                f"Interpreting file '{repr(filepath)}' of component '{module_name}'"
+                " relatively to components directory"
+            )
 
-        # If resolved absolute path does NOT exist or filepath is NOT a string, then return as is
-        logger.debug(
-            f"Interpreting template '{repr(filepath)}' of component '{module_name}'"
-            " relatively to components directory"
+        return resolved_filepaths
+
+    # Check if filepath is a glob pattern that points to files relative to the components directory
+    # (as defined by `COMPONENTS.dirs` and `COMPONENTS.app_dirs` settings) in which the component is defined.
+    # If yes, modify the path to refer to the globbed files.
+    # If not, don't modify anything.
+    def resolve_static_media_file(
+        filepath: Union[str, SafeData],
+        allow_glob: bool,
+    ) -> List[Union[str, SafeData]]:
+        resolved_filepaths, _ = resolve_media_file(
+            filepath,
+            allow_glob,
+            static_files_dir=matched_component_dir_abs,
+            media_root_dir=matched_component_dir_abs,
         )
-        return filepath
+        return resolved_filepaths
 
     # Check if template name is a local file or not
     if getattr(comp_media, "template_file", None):
-        comp_media.template_file = resolve_media_file(comp_media.template_file)
+        comp_media.template_file = resolve_relative_media_file(comp_media.template_file, False)[0]
     if getattr(comp_media, "js_file", None):
-        comp_media.js_file = resolve_media_file(comp_media.js_file)
+        comp_media.js_file = resolve_relative_media_file(comp_media.js_file, False)[0]
     if getattr(comp_media, "css_file", None):
-        comp_media.css_file = resolve_media_file(comp_media.css_file)
+        comp_media.css_file = resolve_relative_media_file(comp_media.css_file, False)[0]
 
     if hasattr(comp_media, "Media") and comp_media.Media:
-        _map_media_filepaths(comp_media.Media, resolve_media_file)
-
-
-def _get_dir_path_from_component_path(
```

Comment:
Renamed `_get_dir_path_from_component_path` to `_find_component_dir_containing_file`, and moved some logic out, so it's clearer what's happening.